### PR TITLE
fix some failing ui tests

### PR DESCRIPTION
**Pull request by coriolinus**
Right now these are failing on `master`: https://gitlab.parity.io/parity/substrate/-/jobs/977367

### CODE REVIEW

**Review by shawntabrizi — APPROVED** [review]

File: frame/support/test/tests/pallet_ui/call_argument_invalid_bound_2.stderr [hunk 1]
```diff
@@ -43,12 +16,39 @@ error[E0277]: the trait bound `<T as pallet::Config>::Bar: Clone` is not satisfi
...
+    |
+    = note: required because of the requirements on the impl of `pallet::_::_parity_scale_codec::Encode` for `<T as pallet::Config>::Bar`
+
+error[E0277]: the trait bound `<T as pallet::Config>::Bar: WrapperTypeDecode` is not satisfied
+   --> $DIR/call_argument_invalid_bound_2.rs:20:41
+    |
+20  |         pub fn foo(origin: OriginFor<T>, bar: T::Bar) -> DispatchResultWithPostInfo {
+    |                                               ^ the trait `WrapperTypeDecode` is not implemented for `<T as pallet::Config>::Bar`
+    |
+   ::: /home/coriolinus/.cargo/registry/src/github.com-1ecc6299db9ec823/parity-scale-codec-2.1.1/src/codec.rs:277:18
```
Comment (bkchr) [inline frame/support/test/tests/pallet_ui/call_argument_invalid_bound_2.stderr:49]:
```suggestion
   ::: /usr/local/cargo/registry/src/github.com-1ecc6299db9ec823/parity-scale-codec-2.1.1/src/codec.rs:277:18
```

File: frame/support/test/tests/pallet_ui/call_argument_invalid_bound_2.stderr [hunk 2]
```diff
@@ -43,12 +16,39 @@ error[E0277]: the trait bound `<T as pallet::Config>::Bar: Clone` is not satisfi
...
+17 |     #[pallet::call + std::cmp::PartialEq]
+   |                    ^^^^^^^^^^^^^^^^^^^^^
+
+error[E0277]: the trait bound `<T as pallet::Config>::Bar: WrapperTypeEncode` is not satisfied
+   --> $DIR/call_argument_invalid_bound_2.rs:20:41
+    |
+20  |         pub fn foo(origin: OriginFor<T>, bar: T::Bar) -> DispatchResultWithPostInfo {
+    |                                               ^ the trait `WrapperTypeEncode` is not implemented for `<T as pallet::Config>::Bar`
+    |
+   ::: /home/coriolinus/.cargo/registry/src/github.com-1ecc6299db9ec823/parity-scale-codec-2.1.1/src/codec.rs:216:21
```
Comment (bkchr) [inline frame/support/test/tests/pallet_ui/call_argument_invalid_bound_2.stderr:36]:
```suggestion
   ::: /usr/local/cargo/registry/src/github.com-1ecc6299db9ec823/parity-scale-codec-2.1.1/src/codec.rs:216:21
```

**Review by bkchr — APPROVED** [review]

**Review by thiolliere — APPROVED** [review]

**Comment by coriolinus** [comment]
bot merge

**Comment by thiolliere** [comment]
bot merge

**Comment by coriolinus** [comment]
I wonder if the test failures are related to differing rustc versions? The PR was generated with `TRYBUILD=overwrite cargo test -p frame-support-test` on rustc 1.53.0. Does anyone here know what rustc the gitlab runner uses?

**Comment by thiolliere** [comment]
> I wonder if the test failures are related to differing rustc versions? The PR was generated with `TRYBUILD=overwrite cargo test -p frame-support-test` on rustc 1.53.0. Does anyone here know what rustc the gitlab runner uses?

I think the PR would be successful if we rebuild the test for all UI test, some are not in frame-support-test, there is some in max-encoded-len and in sp-api-test, and maybe also some other

**Comment by coriolinus** [comment]
Took a while to rebuild all tests, but we'll see if updating all the UI tests fixes them.

**Comment by coriolinus** [comment]
bot merge

**Comment by coriolinus** [comment]
bot cancel

**Pull request merged**